lavalink/player.py

```diff
@@ -119,6 +139,38 @@ def connected(self) -> bool:
         """
         return self._connected
 
+    async def on_voice_server_update(self, data: dict) -> None:
+        self._voice_state.update({"event": data})
+        await self._send_lavalink_voice_update(self._voice_state)
+
+    async def on_voice_state_update(self, data: dict) -> None:
+        self._voice_state.update({"sessionId": data["session_id"]})
+        if (channel_id := data["channel_id"]) is None:
+            ws_rll_log.info("Received voice disconnect from discord, removing player.")
+            self._voice_state.clear()
+            await self.disconnect(force=True)
+        else:
+            channel = self.guild.get_channel(int(channel_id))
+            if channel != self.channel:
+                if self.channel:
+                    self._last_channel_id = self.channel.id
+                self.channel = channel
+        await self._send_lavalink_voice_update({**self._voice_state, "event": data})
+
+    async def _send_lavalink_voice_update(self, voice_state: dict):
+        if voice_state.keys() != {"sessionId", "event"}:
+            return
+
+        if voice_state["event"].keys() == {"token", "guild_id", "endpoint"}:
```

Comment:
We could get rid of this `if` if we simply didn't call this method from `on_voice_state_update`.